kolibri/core/assets/src/utils/syncTaskUtils.js

```diff
@@ -31,24 +69,28 @@ const syncTaskStatusToStepMap = {
   [SyncTaskStatuses.REMOTE_DEQUEUING]: 7,
 };
 
+// These functions are partially applied with bind() so that they are not evaluated before i18n
+// is ready. If you key into this map you'll need to call the value as a fn with no args a la
+// syncStatusToDescriptionMap[SyncTaskStatuses.SESSION_CREATION]();
+// Note that when getTaskString or coreString is called within a function, this is unnecessary.
 const genericStatusToDescriptionMap = {
-  [TaskStatuses.PENDING]: getTaskString('taskWaitingStatus'),
-  [TaskStatuses.QUEUED]: getTaskString('taskWaitingStatus'),
-  [TaskStatuses.COMPLETED]: getTaskString('taskFinishedStatus'),
-  [TaskStatuses.CANCELED]: getTaskString('taskCanceledStatus'),
-  [TaskStatuses.CANCELING]: getTaskString('taskCancelingStatus'),
-  [TaskStatuses.FAILED]: getTaskString('taskFailedStatus'),
+  [TaskStatuses.PENDING]: getTaskString.bind(null, 'taskWaitingStatus'),
```

Comment:
This does indeed work, but I wonder if just an inline anonymous arrow function might be more comprehensible?

`() => getTaskString('taskWaitingStatus')`